### PR TITLE
New App Layout Toolbar

### PR DESCRIPTION
**Pull request by ericdecanini**
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Adds the new app layout toolbar hidden behind the New App Layout Enabled feature flag

## Motivation and context

Closes https://github.com/vector-im/element-android/issues/6580

Figma link: https://www.figma.com/file/Hw4gjP3pknMZUir8jQqmWH/%5BMobile%5D-IA-early-proposals?node-id=2584%3A140337

This does not include behaviour when inside spaces or back navigation

## Screenshots / GIFs

| |Before|After|After (Collapsed)|
|-|-|-|-|
|Light||||
|Dark||||
|Black||||

## Tests

- On a fresh install, access the cog at the bottom of the initial screen, go to Features and enable New App Layout
- Sign in
- See new app toolbar and ensure profile icon and settings show and react (on click) as expected

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): Android 11

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [x] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

**Review by amshakal — APPROVED** [review]
Looking good!

**Comment by kojid0** [comment]
grafik

Can't access figma since it not public, but are you sure that the graphics are correct? Why is there a division into DM and Rooms in the "All Chats" view?

**Comment by ericdecanini** [comment]
You'd be right in thinking the diversion should no longer be there. We're tackling the new app layout bit by bit and hiding it behind the aforementioned feature flag until it's complete. Until then, the app as you see it will not have changed in the slightest

File: library/ui-styles/src/main/res/values/text_appearances.xml [hunk 1]
```diff
@@ -32,6 +32,15 @@
         <item name="android:textColor">?vctr_content_primary</item>
     </style>
 
+    <style name="TextAppearance.Vector.Headline.Bold" parent="TextAppearance.MaterialComponents.Headline1">
```
Comment (fedrunov) [inline library/ui-styles/src/main/res/values/text_appearances.xml:35]: Shouldn't we inherit from existing appearance?

File: vector/src/main/java/im/vector/app/features/crypto/keysbackup/setup/KeysBackupSetupStep3Fragment.kt [hunk 2]
```diff
@@ -142,7 +142,7 @@ class KeysBackupSetupStep3Fragment @Inject constructor() : VectorBaseFragment<Fr
 
         dialog.findViewById<View>(R.id.keys_backup_setup_share)?.debouncedClicks {
             startSharePlainTextIntent(
-                    fragment = this,
+                    context = requireContext(),
```
Comment (fedrunov) [inline vector/src/main/java/im/vector/app/features/crypto/keysbackup/setup/KeysBackupSetupStep3Fragment.kt:145]: where this change comes from?

**Review by fedrunov — CHANGES_REQUESTED** [review]

**Review by fedrunov — APPROVED** [review]
LGTM

**Pull request merged**